hdlparse/vhdl_parser.py

```diff
@@ -6,748 +6,829 @@
 }
 
 VhdlLexer = MiniLexer(vhdl_tokens, flags=re.MULTILINE | re.IGNORECASE)
 
 
 class VhdlObject(object):
-    """Base class for parsed VHDL objects
-
-    Args:
-      name (str): Name of the object
-      desc (str): Description from object metacomments
-    """
-
-    def __init__(self, name, desc=None):
-        self.name = name
-        self.kind = 'unknown'
-        self.desc = desc
+  '''Base class for parsed VHDL objects
 
+  Args:
+    name (str): Name of the object
+    desc (str): Description from object metacomments
+  '''
+  def __init__(self, name, desc=None):
+    self.name = name
+    self.kind = 'unknown'
+    self.desc = desc
 
 class VhdlParameter(object):
-    """Parameter to subprograms, ports, and generics
-
-    Args:
-      name (str): Name of the object
-      mode (str): Direction mode for the parameter
-      data_type (str): Type name for the parameter
-      default_value (str): Default value of the parameter
-      desc (str): Description from object metacomments
-    """
-
-    def __init__(self, name, mode=None, data_type=None, default_value=None, desc=None):
-        self.name = name
-        self.mode = mode
-        self.data_type = data_type
-        self.default_value = default_value
-        self.desc = desc
-
-    def __str__(self):
-        if self.mode is not None:
-            param = '{} : {} {}'.format(self.name, self.mode, self.data_type)
-        else:
-            param = '{} : {}'.format(self.name, self.data_type)
-        if self.default_value is not None:
-            param = '{} := {}'.format(param, self.default_value)
-        return param
-
-    def __repr__(self):
-        return "VhdlParameter('{}', '{}', '{}')".format(self.name, self.mode, self.data_type)
-
+  '''Parameter to subprograms, ports, and generics
+
+  Args:
+    name (str): Name of the object
+    mode (str): Direction mode for the parameter
+    data_type (str): Type name for the parameter
+    default_value (str): Default value of the parameter
+    desc (str): Description from object metacomments
+    param_desc (str): Description of the parameter
+  '''
+  def __init__(self, name, mode=None, data_type=None, default_value=None, desc=None, param_desc = None):
+    self.name = name
+    self.mode = mode
+    self.data_type = data_type
+    self.default_value = default_value
+    self.desc = desc
+    self.param_desc = None
+
+  def __str__(self):
+    if self.mode is not None:
+      param = '{} : {} {}'.format(self.name, self.mode, self.data_type.name + self.data_type.arange)
+    else:
+      param = '{} : {}'.format(self.name, self.data_type.name + self.data_type.arange)
+    if self.default_value is not None:
+      param = '{} := {}'.format(param, self.default_value)
+    if self.param_desc is not None:
+      param = '{} --{}'.format(param, self.param_desc)
+    return param
+
+  def __repr__(self):
+    return "VhdlParameter('{}', '{}', '{}')".format(self.name, self.mode, self.data_type.name + self.data_type.arange)
```

Comment:
It would be good to change `.format(...)` to f-string syntax if possible.
First linters are complaining about `.format(...)` usage.

(Can be done in a new PR. If so, please derive an issue to track the work and the PR.)

Reply:
Forwarded to #5.